lib/crawlerFactory.js

```diff
@@ -206,17 +208,18 @@ class CrawlerFactory {
         factoryLogger.info(`creating refreshingOption config get completed`);
         const defaults = CrawlerFactory.getDefaultOptions();
         return CrawlerFactory.initializeSubsystemOptions(values, defaults[subsystemName]).then(resolved => {
-          factoryLogger.info(`subsystem options initialized`);
+          factoryLogger.info(`subsystem '${subsystemName}' options initialized`);
           result[subsystemName] = values;
         });
       });
     })).then(() => { return result; });
   }
 
   static initializeSubsystemOptions(config, defaults) {
-    if (Object.getOwnPropertyNames(config).length > 1) {
-      return Q(config);
-    }
+    // TODO: This prevents delta values being propagated throughout the factory.
```

Comment:
Did you previously run the crawler using a redis (or other persistent) config store? The idea here is that if the `getAll` in the caller (line 205) gets some values (i.e., there is more than one prop) then we should use them. This is setup to share a config across many instances. If you want to change the config you have to either change them in the store (e.g., redis), clear the store (e.g., redis) or do some crawler API or crawler dashboard calls to change. you can see the current config by starting up the crawler and use the [crawler cli](https://github.com/Microsoft/ghcrawler-cli).

Here is the comment that should appear at line 217 in the original.
```javascript
    // if the incoming config already has values (e.g., retrieved from Redis) then assume it already initialized
    // and do not apply defaults. 
    // TODO: consider a mechanism for forcing flushing config. Would need to ensure this sync'd up well with whatever
    // the persistent config provider is doing.
    // TODO: here we are relying on an implementation detail of the config object that injects a `_config` prop. We 
    // should update the config code to make that prop non enumerable so we don't have to filter it out everywhere.
```

If that is not the problem, can you attach your `env.json` (with the secrets obfuscated) so we can repro?

Reply:
I honestly have no idea what the problem was that I was seeing. I've added the comment to the code.

Originally the defaults were getting swallowed by a config else where, but now it seems fine - so I'm going to assume this was a docker caching issue somewhere and step away.